crml/nft/src/benchmarking.rs

```diff
@@ -0,0 +1,274 @@
+/* Copyright 2019-2021 Centrality Investments Limited
+*
+* Licensed under the LGPL, Version 3.0 (the "License");
+* you may not use this file except in compliance with the License.
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* You may obtain a copy of the License at the root of this project source code,
+* or at:
+*     https://centrality.ai/licenses/gplv3.txt
+*     https://centrality.ai/licenses/lgplv3.txt
+*/
+
+//! NFT benchmarking.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::{account, benchmarks, whitelisted_caller};
+use frame_system::RawOrigin;
+use sp_runtime::Permill;
+
+use crate::types::MAX_ENTITLEMENTS;
+use crate::{Module as Nft, MAX_COLLECTION_ID_LENGTH};
+
+// Create a collection for benchmarking
+// Returns the collection id, schema, and royalties schedule
+fn setup_collection<T: Trait>(creator: T::AccountId) -> (CollectionId, NFTSchema, RoyaltiesSchedule<T::AccountId>) {
+	// Collection id with max length
+	let collection_id = [1_u8; MAX_COLLECTION_ID_LENGTH as usize].to_vec();
+
+	// Schema with max. attributes
+	let schema = (0..MAX_SCHEMA_FIELDS)
+		.map(|i| {
+			let mut v = b"probably-a-decent-upper-length-for-an-attribte-name".to_vec();
+			v.push(i as u8);
+			(v, NFTAttributeValue::String(Default::default()).type_id())
```

Comment:
I would suggest a String longer than Default::default() (which should be empty) for benchmarking. Perhaps 32 bytes long.